admin-client/src/main/java/com/epam/admin_client/config/CustomAuthenticationProvider.java

```diff
@@ -0,0 +1,60 @@
+package com.epam.admin_client.config;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.interfaces.Claim;
+import com.epam.admin_client.service.LoginService;
+import java.io.UnsupportedEncodingException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CustomAuthenticationProvider implements AuthenticationProvider {
+
+    private static final String ADMIN = "ADMIN";
+
+    @Autowired
+    private LoginService loginService;
+
+    @Override
+    public Authentication authenticate(Authentication authentication)
+        throws AuthenticationException {
+
+        String email = authentication.getName();
+        String password = authentication.getCredentials().toString();
+        String token = loginService.authenticate(email, password);
+        JWTVerifier verifier;
+        try {
+            verifier = JWT.require(Algorithm.HMAC256("iteaky"))
```

Comment:
jwt token should only be known in server app (bdd-generator). The verify operation is not required, since we trust remote host ultimately. 
Requirement is as follows: jwt secret key might be changed in config of bdd-generator and existing admin module *should* only require requthentication. (No restart, no reconfig, no recompilation).

Since you do not need to verify the token, the ONLY operation you should call to decode token:

    JWT.decode(token)